packages/core/src/main/self/SelfService.ts

```diff
@@ -24,33 +24,36 @@ import {Self} from '@wireapp/api-client/dist/commonjs/self/index';
 export default class SelfService {
   constructor(private apiClient: APIClient) {}
 
+  public async checkUserName(userName: string): Promise<string | undefined> {
+    const [availableHandle] = await this.checkUserNames([userName]);
+    return availableHandle;
+  }
+
+  public checkUserNames(userNames: string[]): Promise<string[]> {
+    return this.apiClient.user.api.postHandles({
+      handles: userNames,
+    });
+  }
+
   public async getName(): Promise<string> {
     const {name} = await this.apiClient.self.api.getName();
     return name;
   }
 
+  public getSelf(): Promise<Self> {
+    return this.apiClient.self.api.getSelf();
+  }
+
   public async getUserName(): Promise<string | undefined> {
     const {handle} = await this.apiClient.self.api.getSelf();
     return handle;
   }
 
-  public getSelf(): Promise<Self> {
-    return this.apiClient.self.api.getSelf();
-  }
-
   public setName(name: string): Promise<{}> {
     return this.apiClient.self.api.putSelf({name});
   }
 
-  public async setUserName(userName: string): Promise<{}> {
-    const [availableHandle] = await this.apiClient.user.api.postHandles({
-      handles: [userName],
-    });
-
-    if (availableHandle) {
-      return this.apiClient.self.api.putHandle({handle: userName});
-    }
-
-    throw new Error(`Username "${userName}" is not available.`);
+  public setUserName(userName: string): Promise<{}> {
```

Comment:
Let's merge wireapp/wire-web-packages#565 first and then we can remove the carp return type. 🎣